### DISCUSSION TITLE
[Idea] tea packs

### DISCUSSION THREAD

**Discussion by davdroman** [pkgxdev/pkgx#446] [Ideas]
Thinking a bit deeper about the implications of something like #362, I feel like there's a pattern arising that could be generalised as a tea feature.

This comment is especially hinting at it:

> https://github.com/orgs/teaxyz/discussions/362#discussioncomment-4875585
> Beware that you’ll be surprised how often things you don’t expect break if there isn’t POSIX in PATH.

Also:

> https://github.com/orgs/teaxyz/discussions/436#discussioncomment-5280789
> Notably we haven't packaged all of POSIX yet so this will likely leave a somewhat broken shell.

So, if the plan is to package all of POSIX (aka all of `/usr/bin`), could we perhaps consider that a tea "pack"?

```sh
# hypothetical syntax which reuses familiar + prefix
tea +pack/posix +nodejs.org^16 ... # PATH contains all of tea's POSIX, node js & user's PATH
```

It's up for discussion where pack definitions would live. Are they perhaps just locally defined? Maybe they can be declared as part of `package.yml`? Are they vended by tea pantry forks? Not sure, something to explore.

But, with the concept of packs in place, the `--strain` idea becomes a really simple thing to implement and for users to understand. All it does is discard the user's `PATH`! ✨ 

```sh
tea --strain +pack/posix +nodejs.org^16 ... # PATH contains all of tea's POSIX and node js. No user PATH.
```

Am I missing anything here? Is this super naive? I really can't tell but it certainly got me excited when I first thought of it.

**Comment by sschneider-ihre-pvs** [comment]
tea bags, really funny :D but every time I read something like package.* it reminds me of this bloated node_modules folder :(

**Reply by sschneider-ihre-pvs** [reply]
Also, windows

**Comment by mxcl** [comment]
Cool idea. They can just be packages that have `null` for `build` then we would need to update brewkit to skip them.

so `tea +tea.xyz/pack/POSIX` for example.
